### PR TITLE
Support for Rails 5

### PR DESCRIPTION
**Pull request by januszm**
Also see the last commit. Didn't we have a bug in ```def expects``` ? Shouldn't it be a conditional assignment? If not, and if it was intentional (just ```||```) then I'll remove the last commit.

### CODE REVIEW

**Comment by januszm** [comment]
Travis-CI might say it can't merge this because I specified Rails version ```>=4  < 5.1``` in dependencies. This will probably pick 4.2.6 at the moment, instead of 5.0.0.beta3. To make Travis happy we'd need to specify dependency as:

```ruby
  s.add_dependency "rails", ">= 5.0.0.beta3", "< 5.1"
```

File: README.md [hunk 1]
```diff
@@ -129,13 +129,14 @@ end
 
 ## Supported Ruby Versions
 
+* Ruby 2.3.0
 * Ruby 2.0.0
 * Ruby 1.9.3
```
Comment (mynameisrufus) [inline README.md]: Let's drop 1.9 versions :)

File: app/models/proposal/token.rb [hunk 2]
```diff
@@ -62,7 +62,7 @@ def validate_accepted
     scope :reminded, ->{ where('reminded_at IS NOT NULL') }
 
     def expects
-      @expects || proposable.proposal_options[:expects]
+      @expects ||= proposable.proposal_options[:expects]
```
Comment (mynameisrufus) [inline app/models/proposal/token.rb]: I don't see why we would need todo this change, it's not an expensive operation so no point adding?

File: lib/proposal/version.rb [hunk 3]
```diff
@@ -1,3 +1,3 @@
 module Proposal
-  VERSION = "1.0.0"
+  VERSION = "1.0.1"
```
Comment (mynameisrufus) [inline lib/proposal/version.rb]: Let's make this 2.0.0 otherwise people will experience pain with `~> 1.0`

File: README.md [hunk 4]
```diff
@@ -129,13 +129,14 @@ end
...
 * Ruby 1.9.2
 
 ## Supported Rails Versions
```
Comment (mynameisrufus) [inline README.md]: ## Older Rails versions
  
 * Rails 4 `gem 'proposal' ~> '1.0'`
 * Rails 3 `gem 'proposal' ~> '0.0.6'`

File: proposal.gemspec [hunk 5]
```diff
@@ -16,9 +16,7 @@ Gem::Specification.new do |s|
   s.files = Dir["{app,config,db,lib}/**/*"] + ["Rakefile"]
   s.test_files = Dir["test/**/*"]
 
-  s.add_dependency "rails", "~> 4.0"
+  s.add_dependency "rails", ">= 5.0.0.beta3", "< 5.1"
```
Comment (mynameisrufus) [inline proposal.gemspec]: I guess after we merge this PR you could make another later to change this to `~> 5.0`?

File: app/models/proposal/token.rb (hunk 2 above)
Comment (januszm) [inline app/models/proposal/token.rb]: What is the purpose of this operation? did you mean to 'cache with instance variable' here? If yes, then I'd rather use that conditional assignment here.

Just to clarify I grepped for that instance variable name

```bash
(rails-5) proposal: grep -Rin "@expects" *
app/models/proposal/token.rb:65:      @expects ||= proposable.proposal_options[:expects]
```
```@expects``` is not used anywhere else, so this suggests it's only used for caching in #expects method.

File: proposal.gemspec (hunk 5 above)
Comment (januszm) [inline proposal.gemspec]: I've already update this line of code. We can change this to ```~> 5.0``` once it's released, probably in v2.0.1

**Comment by januszm** [comment]
@mynameisrufus I've updated the PR. We stick to Rails 5.x and Ruby 2.x. Who wants to use it with Rails 4 should stick to proposal version 1.x.

**Comment by januszm** [comment]
Added ```rake db:migrate``` to travis.yml to make Travis-CI happy, changed tested Ruby versions to 2.2+ (Rails 5 requirement).

File: app/models/proposal/token.rb (hunk 2 above)
Comment (mynameisrufus) [inline app/models/proposal/token.rb]: I can't remember :( looks good then.

**Comment by mynameisrufus** [comment]
Hey @januszm we lost the actual test run in the `travis.yml`, I fixed it but the tests are actually failing https://travis-ci.org/mynameisrufus/proposal if you could fix then I can push the gem to ruby gems 👍

**Comment by januszm** [comment]
Weird, yesterday I've made all tests passing. It was about:
* limiting Ruby versions to ruby-2.2.2 and ruby-head
* triggering db:migrate since AR now complains loud about pending migrations, in .travis.yml I've added

```script:
    - bundle exec rake db:migrate
```

this might  be ```rails db:migrate``` not ```rake``` actually

**Comment by mynameisrufus** [comment]
Yup adding `db:migrate` nixed the default task `test` see https://github.com/mynameisrufus/proposal/commit/4374b11ed031f65c59953b1028e503cff0a9b0fd

**Comment by januszm** [comment]
I see Travis is already running Rails 5 beta4 released on April 28th and I tested the PR on beta3 . This might be the reason why some tests fail after the PR was merged.

**Pull request merged**